src/Pages/CartPage/CartComponents/CartProduct.js

```diff
@@ -1,37 +1,38 @@
 import React from 'react';
-import { addProductQty, reduceProductQty, removeFromCart, addToWishlist, findProductInList } from '../../../Util/index';
-import { useCart, useWishlist } from '../../../Contexts/index'
+import { Link } from 'react-router-dom'
+import { addProductQty, reduceProductQty, removeFromCart, addToWishlist, handleAddToList, setToast } from '../../../Util/index';
+import { useCart, useWishlist, useToast } from '../../../Contexts/index'
 
 const CartProduct = ({ cartItem }) => {
-    const { title, imgSrc, altTxt, subtitle, qty, reviewsnum, currentprice, discount, originalprice, rating, shortdesc } = cartItem
+    const { title, imgSrc, altTxt, subtitle, qty, reviewsnum, currentprice, discount, originalprice, rating, shortdesc, _id } = cartItem
     const { cartDispatch } = useCart()
     const { wishlistState, wishlistDispatch } = useWishlist()
+    const { dispatchToast } = useToast()
 
     const updateCartList = async (product, getUpdatedCart) => {
         const updatedCart = await getUpdatedCart(product)
         cartDispatch({ type: 'UPDATE_CART', payload: updatedCart })
     }
 
     const moveFromCart = async (cartProduct) => {
-        const newWishlist = []
-        const updatedCart = await removeFromCart(cartProduct)
-        cartDispatch({ type: 'UPDATE_CART', payload: updatedCart })
-        const foundId = findProductInList(cartProduct._id, wishlistState)
-        if (!foundId) {
-            newWishlist = await addToWishlist(cartProduct)
-        } else {
-            console.log("product already in cart. update qty in cart page.")
-            newWishlist = wishlistState
-        }
-        wishlistDispatch({ type: 'UPDATE_WISHLIST', payload: newWishlist })
+        const { updatedList, toastMsg } = await handleAddToList({ addProductFn: addToWishlist, listState: wishlistState, productDetails: cartProduct })
+        console.log('tyty', updatedList, toastMsg)
```

Comment:
This console log is not required ig, you can remove it.

Reply:
done